core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -873,6 +870,13 @@ class CtxSettings : public SettingsBase {
 
             updateActiveParametersImpl();
 
+            // Update sample_rate if the block performs decimation or interpolation
+            if (result.forwardParameters.contains(gr::tag::SAMPLE_RATE.shortKey()) && (_block->input_chunk_size != 1ULL || _block->output_chunk_size != 1ULL)) {
```

Comment:
This 'if' should be guarded by a constexpr-if on the condition that the input->output rate can actually change (i.e. `input_chunk_size` and `output_chunk_size` not being `const` as defined by `ResamplingControl<...>`).